mobile/lib/shared/services/asset.service.dart

```diff
@@ -83,38 +91,30 @@ class AssetService {
   }
 
   /// Returns `null` if the server state did not change, else list of assets
-  Future<List<Asset>?> _getRemoteAssets(User user) async {
+  Future<List<Asset>?> _getRemoteAssets(User user, DateTime until) async {
     const int chunkSize = 10000;
     try {
-      final DateTime now = DateTime.now().toUtc();
       final List<Asset> allAssets = [];
-      for (int i = 0;; i += chunkSize) {
+      DateTime? lastCreationDate;
+      String? lastId;
+      for (;;) {
```

Comment:
`while` loop?

Reply:
I see this a lot in embedded system code, not used to seeing it in higher level language 😝

Reply:
I personally use this for loops when I break out using other means.

While(true).. I use for endless tasks. Like a server listening indefinitely until it's killed etc.

Reply:
Can this be rewritten to a `do while` loop on the condition `assets.length < chunkSize`? I think that's the only break condition I see here. I concur that `while (true)` loops are odd and potentially dangerous. I also think `for (;;)` is less readable. I did not know what it meant until I read some of the other comments.

Reply:
sadly, I can't put the break condition in the loop condition check: it would not even run the first pass of the loop - i can only to while(true) instead of for(;;). The loop has two break conditions that cannot be handled in the loop header